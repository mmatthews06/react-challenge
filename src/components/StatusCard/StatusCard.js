import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  status: {
    fontSize: '1em',
  },
  subTitle: {
    fontSize: '0.75em',
  },
  value: {
    fontSize: '4em',
  },
}));

export default function Status({ label, value }) {
  const classes = useStyles();
  return (
    <Grid container direction="column" spacing={1} justify="center">
      <Grid item>
        <Typography variant="overline" className={classes.status}>
          {label}
        </Typography>
      </Grid>
      <Grid item>
        <Typography variant="subtitle1" className={classes.subTitle}>
          Last 7 days
        </Typography>
      </Grid>
      <Grid item>
        <Typography variant="body1" className={classes.value}>
          {value}
        </Typography>
      </Grid>
    </Grid>
  );
}

Status.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
};
