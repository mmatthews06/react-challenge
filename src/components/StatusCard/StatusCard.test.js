/* global describe, expect, it */
import React from 'react';
import { shallow } from 'enzyme';

import StatusCard from './StatusCard';

describe('StatusCard', () => {
  it('should match a snapshot', () => {
    expect(shallow(<StatusCard label="hello!" value={20} />)).toMatchSnapshot();
  });
});
