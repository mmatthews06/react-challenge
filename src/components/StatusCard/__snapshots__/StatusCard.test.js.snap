// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCard should match a snapshot 1`] = `ShallowWrapper {}`;
